app/concerns/controlled_keywords.rb

```diff
@@ -53,9 +53,9 @@ def set_platform_types
   end
 
   def set_instruments
-    @instruments = get_controlled_keyword_short_names(cmr_client.get_controlled_keywords('instruments').fetch('category', []))
```

Comment:
In my other comment, I mentioned how the places we don't check for a response `success?` or `error?`, may throw an error if the body is a html document. Here, for instance, if the body is a html document and we are on the `acquisition_information` form, `set_platform_types`, `set_instruments`, and `set_projects` will all throw an error (undefined method 'fetch' for String). We should probably change all these methods to check for `success?` or `error?`

Reply:
You're right, I will keep working on it tomorrow. 

The idea was that MMT shouldn't know anything about the CMR responses. MMT shouldn't know where the errors are in the response, it just wants the response to give it some error messages. I think what we should move towards is MMT knowing as little as possible about CMR. 

For instance, when searching collections, `https://cmr.earthdata.nasa.gov/search/collections.umm_json?pretty=true`, MMT doesn't need to know that the collections are at `response.body.fetch('items', [])`. MMT should just be able to ask the response for the items, `response.items`

That is a huge change to the system, so I didn't want to get into it in this ticket, but I'll see if there is an intermediate step I can take within this ticket.

Reply:
I went back and forth on manipulating the body of the response, I don't really have a concrete reason, but I wanted to leave the original response alone as much as I could

Reply:
I think your reasoning makes sense, but I think that means that we should always check the cmr response for `success?` or `error?`, and make sure to set something in the error case that won't cause problems. In this particular file we're not doing that at all, so that's the change I'm suggesting here.

Reply:
In my last commit I searched for `cmr_client.` and made sure every time we are requesting data that we are checking for success or error before looking at the body